app/src/main/java/net/gsantner/opoc/util/StringUtils.java

```diff
@@ -345,6 +350,53 @@ public static void selectLines(final EditText edit, final List<Integer> position
             }
             edit.setSelection(selStart, selEnd);
         }
+        showSelection(edit);
+    }
+
+    // Centre the currently selected text on the screen
+    public static void showSelection(final TextView text) {
+        final Layout layout = text.getLayout();
+        if (layout == null) {
+            return;
+        }
+
+        // Try to find scrolling parents 2 levels up in the hierarchy
+        View hScroll = text, vScroll = text;
+        ViewParent parent1 = text.getParent(), parent2 = parent1.getParent();
+        if (parent1 instanceof ScrollView) {
+            vScroll = (View) parent1;
+            if (parent2 instanceof HorizontalScrollView) {
+                hScroll = (View) parent2;
+            }
+        } else if (parent1 instanceof HorizontalScrollView) {
+            hScroll = (View) parent1;
+            if (parent2 instanceof ScrollView) {
+                vScroll = (View) parent2;
+            }
+        }
+
+        // Find position to set in Y
+        final int[] sel = StringUtils.getSelection(text);
+        final int lineStart = StringUtils.getLineStart(text.getText(), sel[0]);
+
+        final int viewHeight = vScroll.getHeight();
```

Comment:
`getHeight` gets the height in the current state (i.e. reduced when keyboard is open). We need the height in the keyboard closed state.

Reply:
Disregard this, found simpler solution